tpi/templates/main.tf

```
@@ -23,4 +23,10 @@ resource "iterative_machine" "{{name}}" {
 {{ startup_script }}
   EOF
   {% endif %}
+  {% if metadata %}metadata = {
+    {% for key, value in metadata.items() %}
+    {{key}} = "{{value}}"
+    {% endfor %}
+  }
+  {% endif %}
```

isn't this more consistent?

```suggestion
    {% for key, value in metadata.items() %}{{key}} = "{{value}}"{% endfor %}
  }{% endif %}
```

---

also... is it OK to cast everything to string?

---

> _also... is it OK to cast everything to string?_

Yes, it's doubly safe: firstly, it's a `map[string]string` as per [the specification](https://github.com/iterative/terraform-provider-iterative/issues/185#issue-988431303); secondly, Terraform [would automatically convert](https://www.terraform.io/docs/language/expressions/types.html#type-conversion) strings to the target type whenever possible and required.

---

> _isn't this more consistent?_

I'll let you explain that to the HCL2 parser. 😈 

```python
>>> import tpi
>>> print(tpi.render_config(name="machine", cloud="generic", metadata={"key_1": "value_1", "key_2": "value_2"}))
terraform {
  required_providers {
    iterative = {
      source = "iterative/iterative"
    }
  }
}

provider "iterative" {}

resource "iterative_machine" "machine" {
  name = "machine"
  cloud = "generic"









  metadata = {
    key_1 = "value_1"key_2 = "value_2"
  }
}
```

---

```suggestion
    {%- for key, value in metadata.items() %}{{"\n    "}}{{key}} = "{{value}}"{% endfor %}
  }{% endif %}
```

---

this?
```suggestion
    {% for key, value in metadata.items() %}{{key}} = "{{value}}"
    {% endfor %}
  }{% endif %}
```

---

Still leaving an extra newline before the closing curly brace, but looks nice in the template.

```
  metadata = {
    key = "value"

  }
```